### PR TITLE
feat: vault-ssh-helper

### PR DESCRIPTION
**Pull request by loispostula**
## Description of changes

Add [`vault-ssh-helper`](https://github.com/hashicorp/vault-ssh-helper) package: Vault SSH Agent is used to enable one time keys and passwords 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [X] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by loispostula** [comment]
Result of `nixpkgs-review pr 252638` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package blacklisted:</summary>
  <ul>
    <li>nixos-install-tools</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vault-ssh-helper</li>
  </ul>
</details>

**Comment by loispostula** [comment]
Result of `nixpkgs-review pr 252638` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 test built:</summary>
  <ul>
    <li>nixosTests.vault-ssh-helper</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vault-ssh-helper</li>
  </ul>
</details>

File: maintainers/maintainer-list.nix
Comment (emilylange) [inline maintainers/maintainer-list.nix:1]: Please split this change into a separate commit titled `maintainers: add lpostula`.
Meaning, three commits in total:
1. `maintainers: add lpostula`
1. `vault-ssh-helper: init at 0.2.1` 
1. `nixosTests.vault-ssh-helper: init` (or `nixos/tests/vault-ssh-helper: init`)

to comply with the contribution guidelines.
Thanks!

File: nixos/doc/manual/release-notes/rl-2311.section.md [hunk 1]
```diff
@@ -55,6 +55,8 @@
 
 - [NNCP](http://www.nncpgo.org/). Added nncp-daemon and nncp-caller services. Configuration is set with [programs.nncp.settings](#opt-programs.nncp.settings) and the daemons are enabled at [services.nncp](#opt-services.nncp.caller.enable).
 
+- [vault-ssh-helper](https://github.com/hashicorp/vault-ssh-helper). Vault SSH Agent is used to enable one time keys and passwords.
+
```
Comment (emilylange) [inline nixos/doc/manual/release-notes/rl-2311.section.md:59]:
```suggestion
```

This section is for new services, not for new packages.
Please remove :)

**Review by emilylange — CHANGES_REQUESTED** [review]

**Comment by loispostula** [comment]
Rebasing has gone wrong, I'm closing in favour of a new pr.
